src/PrestaShopBundle/EventListener/FilterOrderReturnProductsSearchCriteriaListener.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\EventListener;
+
+use Exception;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnProductException;
+use PrestaShop\PrestaShop\Core\Search\Filters\OrderReturnProductsFilters;
+use PrestaShopBundle\Event\FilterSearchCriteriaEvent;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class FilterOrderReturnProductsSearchCriteriaListener is responsible for updating OrderReturnProducts filter with
+ * order return product id.
+ */
+class FilterOrderReturnProductsSearchCriteriaListener
```

Comment:
Instead of this listener I recommend you to inspire from this filter builder https://github.com/PrestaShop/PrestaShop/blob/c583dc50bb39f7a808fecfa18cfa7192f39d30b4/src/Core/Search/Builder/TypedBuilder/ProductCombinationFiltersBuilder.php#L43

It's the same principle you need the parent ID to be in the filters/searchcriteria values, but using `TypedFiltersBuilderInterface` is clear it's part of the parameter resolver feature, and you can unit test your custom builder more easily. It also enforces the fact that `OrderReturnProductsFilters` is not valid unless it has the parent ID defined

Reply:
I don't see this being used at all, I guess at some point I implemented using this and then forgot to remove it. 
Because filtering works without this and there is no search(I think it's not really needed)